### ISSUE TITLE
Diskswapping / Usage with more than 4 disks

### ISSUE THREAD

**Issue by zach-morris** [libretro/libretro-uae#180]
This may be more of a question on usage, but I'm wondering how best to work with games with more than 4 disks, although I would request the possibility of modifying uae to accept more than 4 - I'm not sure if thats an Amiga OS limitation or not?

For example 'Beneath a Steel Sky (Europe)' contains 15 disks.  I can create a uae file with the following:
```
unix.floppy_path=/path_to/Beneath a Steel Sky (Europe)/
nr_floppies=4
floppy0=Beneath a Steel Sky (Europe) (Boot Disk).ipf
floppy1=Beneath a Steel Sky (Europe) (Disk 00).ipf
floppy2=Beneath a Steel Sky (Europe) (Disk 01).ipf
floppy3=Beneath a Steel Sky (Europe) (Disk 02).ipf
diskimage0=Beneath a Steel Sky (Europe) (Disk 03).ipf
diskimage1=Beneath a Steel Sky (Europe) (Disk 04).ipf
diskimage2=Beneath a Steel Sky (Europe) (Disk 05).ipf
diskimage3=Beneath a Steel Sky (Europe) (Disk 06).ipf
diskimage4=Beneath a Steel Sky (Europe) (Disk 07).ipf
diskimage5=Beneath a Steel Sky (Europe) (Disk 08).ipf
diskimage6=Beneath a Steel Sky (Europe) (Disk 09).ipf
diskimage7=Beneath a Steel Sky (Europe) (Disk 10).ipf
diskimage8=Beneath a Steel Sky (Europe) (Disk 11).ipf
diskimage9=Beneath a Steel Sky (Europe) (Disk 12).ipf
diskimage10=Beneath a Steel Sky (Europe) (Disk 13).ipf
```

With the floppy0-3 being the disks available immediately after launching the game, and the diskimage0-10 available as swappable disks while running.  Although in the emulator, I'm not sure how to swap disks if necessary, is there a hotkey or other way to do this?

The other alternative is to create an m3u file containing:
```
Beneath a Steel Sky (Europe) (Boot Disk).ipf
Beneath a Steel Sky (Europe) (Disk 00).ipf
Beneath a Steel Sky (Europe) (Disk 01).ipf
Beneath a Steel Sky (Europe) (Disk 02).ipf
Beneath a Steel Sky (Europe) (Disk 03).ipf
Beneath a Steel Sky (Europe) (Disk 04).ipf
Beneath a Steel Sky (Europe) (Disk 05).ipf
Beneath a Steel Sky (Europe) (Disk 06).ipf
Beneath a Steel Sky (Europe) (Disk 07).ipf
Beneath a Steel Sky (Europe) (Disk 08).ipf
Beneath a Steel Sky (Europe) (Disk 09).ipf
Beneath a Steel Sky (Europe) (Disk 10).ipf
Beneath a Steel Sky (Europe) (Disk 11).ipf
Beneath a Steel Sky (Europe) (Disk 12).ipf
Beneath a Steel Sky (Europe) (Disk 13).ipf
```
However, then only the first disk is available to the emulator and swapping disks is needed via the RA gui.

So, long story short, is there a way to have all the disks available to the emulator on launch?  Or would it be possible to increase nr_floppies to a higher number in the emulator, or allow diskswapping with the diskimages in the uae config file via hotkey/virtual keyboard.

Thanks in advance and for all the work on this emulator

**Comment by sonninnos** [comment]
No, Amiga only allows 3 external floppy drives, so the right approach is to install such a huge game to a HD and/or use the WHDLoad version.

That `diskimage=` is for the UAE GUI's own swapping system, which this core skips completely in favor of the M3U parsing. And it would be quite a mouthful to decide which disk should be inserted into which drive and when.. At least with this current RA's Disk Control interface.

**Comment by zach-morris** [comment]
OK, thanks for the info.  I'd love to use WHDLoad, but it currently segfaults on OSX.

**Comment by sonninnos** [comment]
Let's hope that some OSX wizard joins to help with that.
